dependabot/update_lang_version.py

```diff
@@ -82,26 +93,119 @@ def get_module_list_json():
+            modules_list[idx][MODULE_PR_CHECK_STATUS] = "timed out"
+            break
+
+    all_prs_merged = True
+    for module in pr_passed_modules:
+        repo = github.get_repo(ORGANIZATION + "/" + module[MODULE_NAME])
+        pr = repo.get_pull(module[MODULE_CREATED_PR].number)
+        if (module[MODULE_AUTO_MERGE] & ("AUTO MERGE" in pr.title)):
+            try:
+                pr.merge()
+            except:
+                print ("Error occurred while merging dependency PR for module '" + module[MODULE_NAME] + "'", e)
+                all_prs_merged = False
+
+    if (len(pr_passed_modules) != totalModules):
+        if len(pr_failed_modules) > 0:
+            print ("Following modules dependency PRs have failed checks...")
+            for module in pr_failed_modules:
+                print (module[MODULE_NAME])
+                for check in module[MODULE_FAILED_PR_CHECKS]:
+                    print("[" + module[MODULE_NAME] + "] PR check '" + check["name"] + "' failed for " + check["html_url"])
+        if len(modules_list) > 0:
+            print ("Following modules dependency PRs check validation has timed out...")
+            for module in modules_list:
+                print (module[MODULE_NAME])
+        sys.exit(1)
+
+    if (not all_prs_merged):
+        sys.exit(1)
 
+def check_pending_pr_checks(github, modules_list, index, pr_passed_modules, pr_failed_modules):
+    print("[" + modules_list[index][MODULE_NAME] + "] Checking status of the PR...")
+    passing = True
+    pending = False
+    repo = github.get_repo(ORGANIZATION + "/" + modules_list[index][MODULE_NAME])
+    sha = repo.get_pull(modules_list[index][MODULE_CREATED_PR].number).head.sha
+
+    failed_pr_checks = []
+    for pr_check in repo.get_commit(sha=sha).get_check_runs():
+        if pr_check.conclusion == "success":
+            continue
+        elif pr_check.conclusion == "failure":
+            failed_pr_check = {
+                "name": pr_check.name,
+                "html_url": pr_check.html_url
+            }
+            failed_pr_checks.append(failed_pr_check)
+            passing = False
+        else:
+            pending = True
+            break
 
-def update_module(module_name, lang_version):
-    github = Github(packagePAT)
-    repo = github.get_repo(ORGANIZATION + "/" + module_name)
+    if (not pending):
+        if passing:
+            modules_list[index][MODULE_PR_CHECK_STATUS] = "success"
+            pr_passed_modules.append(modules_list[index])
+        else:
+            modules_list[index][MODULE_PR_CHECK_STATUS] = "failure"
+            modules_list[index][MODULE_FAILED_PR_CHECKS] = failed_pr_checks
+            pr_failed_modules.append(modules_list[index])
+        del(modules_list[index])
+
+def update_module(github, module, lang_version):
+    repo = github.get_repo(ORGANIZATION + "/" + module[MODULE_NAME])
     try:
         properties_file = repo.get_contents(PROPERTIES_FILE, ref=LANG_VERSION_UPDATE_BRANCH)
```

Comment:
shouldn't we always pick up from master?

Reply:
Fixed in 74ab50a